openmsupply/conforma-server
# #769 /preview-action endpoint

Fix #769 

Implements as per the [agreed solution](https://github.com/openmsupply/conforma-web-app/issues/1203#issuecomment-1169449868)

See inline comments for additional details.

Edit: this now includes the file "description" value, which closes #776 as well

### To test:

Load snapshot in template repo: `/dev/snapshots/previewDocsTesting_2022-07-01`

Have changed the "DEV_TEST" triggers in the last PR to "ON_PREVIEW" triggers and added a document that gets generated.

You can see the resulting data returned (and doc created) when either approving or rejecting using Postman:

POST request: `http://localhost:8080/api/preview-actions` (JWT authorization as normal)

To reject, use this body JSON:
```
{
    "reviewId": 3,
    "previewData": {
        "outcome": "REJECTED"
    }
}
```

To approve, use this:
```
{
    "reviewId": 3,
    "previewData": {
        "outcome": "APPROVED"
    }
}
```

This overrides the current "outcome" value (PENDING) with the simulated one, which is the condition on which either the approve or reject notifications go out.
